Src/Extern/SENSEI/AMReX_AmrParticleInSituBridge.H

```diff
@@ -0,0 +1,100 @@
+#ifndef AMReX_AmrParticleInSituBridge_H
+#define AMReX_AmrParticleInSituBridge_H
+#include <AMReX_Config.H>
+
+#ifdef AMREX_PARTICLES
+#include <AMReX_InSituBridge.H>
+#include <AMReX_Vector.H>
+
+#include <AMReX_AmrParticles.H>
+
+
+#include <AMReX_AmrParticleDataAdaptor.H>
+#include <AMReX_AmrDataAdaptor.H>
+
+#include <AMReX_Amr.H>
+
+#ifdef BL_USE_SENSEI_INSITU
+#include <AnalysisAdaptor.h>
+#include <Profiler.h>
+#endif
+
+#include <chrono>
+
+namespace amrex
+{
+/// Bridge for simulations that use amrex::ParticleContainer
+/// inherits intialize and finalize from amrex::InSituBridge
+template<int NStructReal, int NStructInt, int NArrayReal, int NArrayInt>
+class AmrParticleInSituBridge : public InSituBridge
+{
+public:
+  AmrParticleInSituBridge() {}
+  ~AmrParticleInSituBridge() {}
+
+  AmrParticleInSituBridge(const AmrParticleInSituBridge&) = delete;
+  void operator=(const AmrParticleInSituBridge&) = delete;
+
+  // invoke the in situ analysis with data from an AmrMesh
+  // instance.
+  int update(
+    amrex::Amr *amr,
```

Comment:
This bridge does not have the capability of visualizing the data in the amrex::Amr object. We are are getting time & time step from the passed instance. Instead of an amrex::Amr instance we can pass time and time step directly, so the bridge need not depend on amrex::Amr. Note, in some of the particle tutorials for example amrex::Amr is not used.